lib/reek/smells/smell_repository.rb

```diff
@@ -43,25 +42,21 @@ def examine(scope)
       end
 
       def detectors
-        @initialized_detectors ||= begin
-          @detectors = {}
-          smell_types.each do |klass|
-            @detectors[klass] = klass.new(source_via)
-          end
-          @detectors
-        end
+        @initialized_detectors ||= smell_types.map do |klass|
+          { klass => klass.new(source_via) }
+        end.reduce({}, :merge)
       end
 
-      private
+      protected
```

Comment:
Strong veto. We just do this to silence a warning we only see when we run the spec?
Sorry, but for me this is a bad change. SmellRepository does not have any children. Making this protected is highly misleading then since I'd naturally assume it does.